### ISSUE TITLE
FileResponse generated for Task<ActionResult> method

### ISSUE THREAD

**Issue by Entroper** [RicoSuter/NSwag#2982]
I have this method:

		[HttpDelete("{thingId}")]
		public async Task<ActionResult> DeleteThing(int thingId)
		{
			if (await _thingService.DeleteThing(thingId) == 0)
				return NotFound();
			else
				return Ok();
		}

Which generates this Swagger:

        "operationId": "Things_DeleteThing",
        "parameters": [
          {
            "type": "integer",
            "name": "thingId",
            "in": "path",
            "required": true,
            "format": "int32",
            "x-nullable": false
          }
        ],
        "responses": {
          "200": {
            "x-nullable": true,
            "description": "",
            "schema": {
              "type": "file"
            }
          }
        }

And of course the TypeScript client method returns `Observable<FileResponse | null>`.  I'm basically just trying to return 200 if the thing was deleted, 404 if it wasn't found.

Is there another way to achieve an empty result type?  I basically want an `Observable<void>` that errors if the thing wasn't found.

**Comment by syndicatedshannon** [comment]
I would disagree that ActionResult and EmptyResult imply OpenAPI "file" type. In my opinion they correspond better to Any and null. I use the following operation processor. It's inconvenient, because it's the only thing preventing me from using out-of-box features during OpenAPI document generation. I have to create an intermediate project for this operation processor for every client project or add NSwag libraries directly into my API library which doesn't comply with my security and maintenance procedures. Anyway, here's the code, hope it helps you:

```
private class FixActionResult : IOperationProcessor
{
    public bool Process(OperationProcessorContext op)
    {
        var type = ResultType(op.MethodInfo);
        var response = op.OperationDescription.Operation.Responses["200"];
        if (HasType<ActionResult>(type))
        {
            response.Schema = JsonSchema.CreateAnySchema();
        }
        else if (HasType<EmptyResult>(type))
        {
            response.Schema = null;
        }
        return true;
    }
    private Type ResultType(MethodInfo method)
    {
        return method
            .GetCustomAttributes(false)
            .OfType<ProducesResponseTypeAttribute>()
            .SingleOrDefault(s => s.StatusCode == 200)?.Type
            ?? method.ReturnType;
    }
    private bool HasType<T>(Type type)
    {
        return type == typeof(T) || type == typeof(Task<T>);
    }
}
```

**Comment by RicoSuter** [comment]
> I would disagree that ActionResult and EmptyResult imply OpenAPI "file" type. In my opinion they correspond better to Any and null. I use the following operation processor.

ActionResult/EmptyResult is really an untyped/unknown response type so the best way to say any is a file response (= can be anything). You need to provide a response type so that this goes away..

**Comment by syndicatedshannon** [comment]
I think we agree about ActionResult.

EmptyResult we probably disagree on. I would agree that if EmptyResult returned the values "null" or "{}" it would correspond to any. But EmptyResult and void return the same content. Note that the ability to return an EmptyResult from ActionResult doesn't mean that EmptyResult is any.

Anyway, there are multiple workarounds, so no issue for my needs if we disagree a little. NSwag has been very useful, and I've depended on it for a long time. Thank you!
